AutoGraph/ResponseHandler.swift

```diff
@@ -35,23 +32,21 @@ open class ResponseHandler {
             }
     }
     
-    private func map<_MappingKey, _Mapping, _KeyCollection, _ThreadAdapter>(
-        json: JSONValue,
-        objectBinding: ObjectBinding<_MappingKey, _Mapping, _KeyCollection, _ThreadAdapter>) {
+    private func map<SerializedObject: Decodable>(json: JSONValue, objectBinding: ObjectBinding<SerializedObject>) {
             
             do {
                 switch objectBinding {
-                case .object(let binding, let threadAdapter, let keys, let completion):
-                    let mapper = Mapper()
-                    let result: _Mapping.MappedObject = try mapper.map(from: json, using: binding(), keyedBy: keys)
+                case .object(let keyPath, let completion):
                     
-                    if let threadAdapter = threadAdapter {
-                        self.refetchAndComplete(result: result, json: json, mapping: binding, threadAdapter: threadAdapter, completion: completion)
+                    guard let objectJson = json[keyPath] else {
+                        throw AutoGraphError.mapping(error: nil)
```

Comment:
I wasn't sure what do here in the case where we're unable to get an object at the specified key path for the request. It feels like a mapping error but there's no access to an underlying error to pass through here, so I just put in `nil` here to satisfy the compiler. Open to suggestions.

Reply:
Suggestion would be to write an error with an informative string on the fly and pass that to `.mapping(error: error)`. You could look at Crust for some inspiration.

Reply:
fyi, whatever is thrown here is caught and goes into a mapping error anyway, so just throw an error. no reason to wrap it at this juncture.

Reply:
`Error` is a protocol so what I did was just add that conformance to `String` and threw the informative string itself, which will then be wrapped in the mapping error. Cool?